src/engine/renderer/gl_shader.cpp

```diff
@@ -1039,10 +1099,8 @@ void GLShaderManager::BindAttribLocations( GLuint program ) const
 
 bool GLCompileMacro_USE_VERTEX_SKINNING::HasConflictingMacros( size_t permutation, const std::vector< GLCompileMacro * > &macros ) const
 {
-	for ( size_t i = 0; i < macros.size(); i++ )
+	for (const GLCompileMacro* macro : macros)
```

Comment:
I normally do that too. But I didn't like the idea of looping with references to pointers rather than pointers. And auto on it's own risks copying larger things if someone changes the loop.
I'll leave this for now as I think we want to get rid of much of this code anyway. A vector of pointers isn't ideal. Unique and shared ptr  are probably more suitable here but I'll wait until I understand that code to refactor to that and we discuss things in that context.